ds-caselaw-ingester/tests.py

```diff
@@ -842,3 +799,89 @@ def test_unquote_s3(self, getenv, os):
 @pytest.mark.parametrize("was, now", modify_filename_data)
 def test_modify_targz_filename(was, now):
     assert lambda_function.modify_filename(was, addition="_") == now
+
+
+class TestPublicationLogic:
+    def test_v2_ingest_publish(self, v2_ingest):
+        assert v2_ingest.will_publish() is False
+
+    def test_s3_ingest_publish(self, s3_ingest):
+        assert s3_ingest.will_publish() is True
+
+    def test_s3_ingest_publish_no_force_publish(self, s3_ingest):
+        with patch(
+            "lambda_function.Metadata.force_publish", new_callable=PropertyMock
+        ) as mock:
+            mock.return_value = False
+            assert s3_ingest.will_publish() is False
+
+    @patch("lambda_function.api_client.get_published", return_value=False)
+    def test_fcl_not_published_if_not_published(self, get_published, fcl_ingest):
+        assert fcl_ingest.will_publish() is False
+
+    @patch("lambda_function.api_client.get_published", return_value=True)
+    def test_fcl_published_if_published(self, get_published, fcl_ingest):
+        assert fcl_ingest.will_publish() is True
+
+
+class TestEmailLogic:
+    def test_v2_ingest_publish_email_update(self, v2_ingest):
+        v2_ingest.inserted = False
+        v2_ingest.updated = True
+
+        v2_ingest.send_updated_judgment_notification = MagicMock()
+        v2_ingest.send_new_judgment_notification = MagicMock()
+        v2_ingest.send_bulk_judgment_notification = MagicMock()
+        v2_ingest.send_email()
```

Comment:
Worth adding a new line so we are following arrange, act, assert. Same for below tests.